pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -16,266 +16,290 @@ package extensionresources
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&ContainerRuntimeValidator{}).For(&extensionsv1alpha1.ContainerRuntime{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&ControlPlaneValidator{}).For(&extensionsv1alpha1.ControlPlane{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&ContainerRuntimeValidator{}).For(&extensionsv1alpha1.ContainerRuntime{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&DNSRecordValidator{}).For(&extensionsv1alpha1.DNSRecord{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&EtcdValidator{}).For(&druidv1alpha1.Etcd{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&ExtensionValidator{}).For(&extensionsv1alpha1.Extension{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&InfrastructureValidator{}).For(&extensionsv1alpha1.Infrastructure{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&NetworkValidator{}).For(&extensionsv1alpha1.Network{}).Complete()
+	err = builder.WebhookManagedBy(mgr).WithValidator(&OperatingSystemConfigValidator{}).For(&extensionsv1alpha1.OperatingSystemConfig{}).Complete()
+	if err != nil {
+		return err
+	}
+	err = builder.WebhookManagedBy(mgr).WithValidator(&WorkerValidator{}).For(&extensionsv1alpha1.Worker{}).Complete()
+	if err != nil {
+		return err
 	}
 
-	return &h
+	return nil
 }
 
-type handler struct {
-	decoder                        *admission.Decoder
-	logger                         logr.Logger
-	artifacts                      map[metav1.GroupVersionResource]*artifact
-	allowInvalidExtensionResources bool
+var _ admission.CustomValidator = &BackupBucketValidator{}
+var _ admission.CustomValidator = &BackupEntryValidator{}
+var _ admission.CustomValidator = &BastionValidator{}
+var _ admission.CustomValidator = &ContainerRuntimeValidator{}
+var _ admission.CustomValidator = &ControlPlaneValidator{}
+var _ admission.CustomValidator = &DNSRecordValidator{}
+var _ admission.CustomValidator = &EtcdValidator{}
+var _ admission.CustomValidator = &ExtensionValidator{}
+var _ admission.CustomValidator = &InfrastructureValidator{}
+var _ admission.CustomValidator = &NetworkValidator{}
+var _ admission.CustomValidator = &OperatingSystemConfigValidator{}
+var _ admission.CustomValidator = &WorkerValidator{}
+
+type BackupBucketValidator struct{}
+type BackupEntryValidator struct{}
+type BastionValidator struct{}
+type ContainerRuntimeValidator struct{}
+type ControlPlaneValidator struct{}
+type DNSRecordValidator struct{}
+type EtcdValidator struct{}
+type ExtensionValidator struct{}
+type InfrastructureValidator struct{}
+type NetworkValidator struct{}
+type OperatingSystemConfigValidator struct{}
+type WorkerValidator struct{}
```

Comment:
Nit: Use `var ( ) ` and `struct ( )` blocks?

Reply:
Not clear about struct block, can you explain what is needed? is it to create one more struct and wrap all struct in that?

Reply:
Sorry, I meant `type ()` block, of course, just to improve readability a bit:

```golang
var (
    _ admission.CustomValidator = &BackupBucketValidator{}
    _ admission.CustomValidator = &BackupEntryValidator{}
    // ...
)

type (
    BackupBucketValidator struct{}
    BackupEntryValidator  struct{}
    // ...
)
```